VectorTest.cpp

```diff
@@ -0,0 +1,13 @@
+#include "TestHarness.h"
+
+#include <vector>
+
+TEST(vectorIntEquality, vector)
+{
+	std::vector<int> testVector = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+	for (auto index : testVector)
```

Comment:
This is pretty muddled here.  Let me rewrite it to see why:
```
std::vector<int> testVector = { 1, 1, 2, 3, 5, 8, 13, 21, 34, 55 };
	for (auto item : testVector)
	{
		CHECK_EQUAL(item, testVector.at(item));
	}
```
By using the range-based `for` loop, you're already getting away from any indexing into the array, and are just pulling out each value in sequence and operating on it.

So first off, `index` is a really confusing name to give the current item from the vector that is being processed.

Then, if you give the array a set of different values, you can see that what you're doing with `at()` only works for the special case where the _n_th item always has value _n_.  

With the new set of values, the test passes on the first item, but only by coincidence, because both testVector[0] and testVector[1] are equal to 1. The test would pass the second, third, and fourth times through. because the index == the value at that index.  Our luck runs out on the 5th time through, because 5 != testVector[5] (which is 8).  By the time we get to testVector.at(13), we're already off the end of the 10-item vector.

Probably clearer in this case to use a non-range-based `for` loop, so that you can use the actual loop index as the expected value to test against.